shared/macros-bash.jinja

```diff
@@ -107,6 +107,55 @@ if test -x /usr/sbin/prelink; then
 fi
 {{%- endmacro -%}}
 
+{{#
+# Macro to configure DConf settings for RHEL and Fedora systems.
+#}}
+{{%- macro dconf_settings(path, key, value, db, setting_file) -%}}
+# Check for setting in any of the DConf db directories
+# If files contain ibus or distro, ignore them.
+# The assignment assumes that individual filenames don't contain :
+readarray -t SETTINGSFILES < <(grep -r "\\[{{{ path }}}\\]" "/etc/dconf/db/" | grep -v "distro\\|ibus" | cut -d":" -f1)
+DCONFFILE="/etc/dconf/db/{{{ db }}}/{{{ setting_file }}}"
+DBDIR="/etc/dconf/db/{{{ db }}}"
+
+mkdir -p "${DBDIR}"
+
+if [ ${#SETTINGSFILES[@]} -eq 0 ]
```

Comment:
Please add quoting to variable expansion, so the code is somewhat more robust and we are not spammed by `shellcheck` later.